src/nbl/ext/MitsubaLoader/CMitsubaMaterialCompilerFrontend.cpp

```diff
@@ -57,42 +57,54 @@ auto CMitsubaMaterialCompilerFrontend::getTexture(const CElementTexture* _elemen
     if (!viewBundle.getContents().empty())
     {
         auto view = core::smart_refctd_ptr_static_cast<asset::ICPUImageView>(viewBundle.getContents().begin()[0]);
-
         types[0] = asset::IAsset::ET_SAMPLER;
-        const std::string samplerKey = m_loaderContext->samplerCacheKey(m_loaderContext->emissionProfileSamplerParams(*_element));
+        const auto& meta = *static_cast<const asset::CIESProfileMetadata*>(viewBundle.getMetadata());
+        const std::string samplerKey = m_loaderContext->samplerCacheKey(m_loaderContext->emissionProfileSamplerParams(*_element, meta));
         auto samplerBundle = m_loaderContext->override_->findCachedAsset(samplerKey, types, m_loaderContext->inner, 0u);
         assert(!samplerBundle.getContents().empty());
         auto sampler = core::smart_refctd_ptr_static_cast<asset::ICPUSampler>(samplerBundle.getContents().begin()[0]);
 
-        return { view, sampler, 1.0f };
+        return { view, sampler, &meta };
     }
-    return { nullptr, nullptr, 1.0f };
+    return { nullptr, nullptr, nullptr };
 }
 
 CMitsubaMaterialCompilerFrontend::EmitterNode* CMitsubaMaterialCompilerFrontend::createEmitterNode(asset::material_compiler::IR* ir, const CElementEmitter* _emitter, core::matrix4SIMD transform) {
     assert(_emitter->type == CElementEmitter::AREA);
     auto* res = ir->allocNode<asset::material_compiler::IR::CEmitterNode>();
+    res->intensity = _emitter->area.radiance;
     if (_emitter->area.emissionProfile) {
         asset::material_compiler::IR::CEmitterNode::EmissionProfile profile;
-        asset::material_compiler::IR::INode::STextureSource tex;
-        std::tie(tex.image, tex.sampler, tex.scale) = getEmissionProfile(_emitter->area.emissionProfile);
-        assert(tex.image && "emission profile not loaded");
-        profile.texture = tex;
-        profile.normalizeEnergy = _emitter->area.emissionProfile->normalizeEnergy;
-        auto inverseTransform = core::transpose(core::concatenateBFollowedByA(transform, _emitter->area.emissionProfile->transform.matrix));
-        profile.left = inverseTransform[0];
-        profile.up = inverseTransform[1];
+        auto [image, sampler, meta] = getEmissionProfile(_emitter->area.emissionProfile);
+        if (!image) {
+            os::Printer::log("ERROR: Emission profile not loaded", ELL_ERROR);
+            return nullptr;
+        }
+        profile.texture = { image, sampler, 1.f };
+
+        auto inverseTransform = core::transpose(core::concatenateBFollowedByA(transform, _emitter->transform.matrix));
+        profile.right_hand = core::determinant(inverseTransform) >= 0.0f;
+        profile.up = core::normalize(inverseTransform[1]);
+        profile.view = core::normalize(inverseTransform[2]);
+
         res->emissionProfile = profile;
+        float normalizeEnergy = _emitter->area.emissionProfile->normalizeEnergy;
+        if (normalizeEnergy < 0.0f) {
+            res->intensity *= meta->getMaxIntensity();
```

Comment:
normalization implies division though?

Reply:
wasn't that what you wanted?

Reply:
This is the "max mode" implementation. Normalization by energy is done in the next line.

Reply:
ok I've had to reheat this a bit in my brain cause I lost the thread a bit.

> there are 3 modes you should support:
> 
>     * no normalization (default)
> 
>     * normalize energy
> 
>     * normalize Max
> 
> 
> you can use 0 to mean no normalization, positive values for energy and any negative for max

but you load the IES as a UNORM which needs to be multiplied with the max value so that you recover the original values.

So:
no normalization [zero] => `intensity * meta->getMaxIntensity()`
normalization by Max [negative] => `intensity` (leave alone, don't do anything
normalization by Energy => `intensity * meta->getMaxIntensity()/meta->getIntegral()`